lib/rspec/core/formatters/bisect_progress_formatter.rb

```diff
@@ -6,15 +6,6 @@ module Formatters
       # @private
       # Produces progress output while bisecting.
       class BisectProgressFormatter < BaseTextFormatter
-        # We've named all events with a `bisect_` prefix to prevent naming collisions.
-        Formatters.register self, :bisect_starting, :bisect_original_run_complete,
-                            :bisect_round_started, :bisect_individual_run_complete,
-                            :bisect_complete, :bisect_repro_command,
-                            :bisect_failed, :bisect_aborted,
-                            :bisect_round_ignoring_ids, :bisect_round_detected_multiple_culprits,
-                            :bisect_dependency_check_started, :bisect_dependency_check_passed,
-                            :bisect_dependency_check_failed
```

Comment:
not sure I follow why all this goes away?

Reply:
I think you answered your own question above, but for posterity: previously, this was a "real" formatter that used with the `RSpec::Core::Reporter`, which required we register the formatter with all the events it subscribes to.

Now that we are no longer using the reporter to publish events to the formatter during bisection runs, we don't have to register all these events.  Instead, the new `Notifier` I wrote just checks `respond_to?` as a way to see if it can safely publish an event.

In general, we've learned that `respond_to?` isn't a good approach with 3rd-party formatters (particularly when paired with common event names like `start` or `close` that could be meant for some entirely different purpose, or monkey patched onto all objects by a 3rd-party gem, etc...), but when we entirely control the formatter, as we do here, `respond_to?` is simple and efficient.